provider/docker.go

```diff
@@ -186,21 +189,82 @@ func (p *Docker) Init() error {
 			utils.Save(p.Cfg.DataVolume+"/application.json", p.Apps)
 		}
 	}
+
+	// Start all persistent apps if not already running
+	if _, err := os.Stat(p.Cfg.DataVolume+"/application_pimages"); !os.IsNotExist(err) {
+		pfiles, _ := ioutil.ReadDir(p.Cfg.DataVolume+"/application_pimages")
+		for _, pfile := range pfiles {
+			// For each item in directory make sure we have a NAME.tar.gz and NAME.json pair to process
+			if !pfile.IsDir() && strings.HasSuffix(pfile.Name(), ".tar.gz") {
+				pName := strings.TrimSuffix(pfile.Name(), ".tar.gz")
+				if _, err := os.Stat(p.Cfg.DataVolume+"/application_pimages/"+pName+".json"); !os.IsNotExist(err) {
+					// Record the persistent app for future reference
+					var m types.Metadata; utils.Load(p.Cfg.DataVolume+"/application_pimages/"+pName+".json", &m)
+					p.PApps[pName] = &m
+
+					// See if persistent app name is already available in running apps
+					deployPersisApp := true
+					for id := range p.Apps {
+						if p.Apps[id].Info.Name == pName {
+							deployPersisApp = false
+							break
+						}
+					}
+
+					// If app is not already running then start it.
+					if deployPersisApp {
+						f, err := os.Open(p.Cfg.DataVolume+"/application_pimages/"+pfile.Name())
+						if err != nil {
+							return err
+						}
+						p.Deploy(m, f, false)
+					}
+				} else {
+					err = errors.New("Persistent image "+pName+" is missing metadata json")
+				}
+			}
+		}
+	}
+
 	NewListener(p)
 	return nil
 }
 
 // Deploy ...
-func (p *Docker) Deploy(metadata types.Metadata, file io.Reader) (*types.App, error) {
+func (p *Docker) Deploy(metadata types.Metadata, file io.Reader, persistant bool) (*types.App, error) {
```

Comment:
Single case of persistent misspelled as 'persistant'; not significant enough to hold up approval.